packages/multichain/src/caip25Permission.ts

```diff
@@ -0,0 +1,283 @@
+import type { NetworkClientId } from '@metamask/network-controller';
+import type {
+  PermissionSpecificationBuilder,
+  EndowmentGetterParams,
+  ValidPermissionSpecification,
+  PermissionValidatorConstraint,
+  PermissionConstraint,
+} from '@metamask/permission-controller';
+import {
+  CaveatMutatorOperation,
+  PermissionType,
+} from '@metamask/permission-controller';
+import type { CaipAccountId, Json } from '@metamask/utils';
+import {
+  parseCaipAccountId,
+  type Hex,
+  type NonEmptyArray,
+} from '@metamask/utils';
+import { cloneDeep, isEqual } from 'lodash';
+
+import { getEthAccounts } from './adapters/caip-permission-adapter-eth-accounts';
+import { assertScopesSupported } from './scope/assert';
+import { validateAndNormalizeScopes } from './scope/authorization';
+import type {
+  ExternalScopeString,
+  InternalScopeObject,
+  InternalScopesObject,
+} from './scope/types';
+
+export type Caip25CaveatValue = {
```

Comment:
Nit: Would be nice to have docs for these exported types

Reply:
added here: https://github.com/MetaMask/core/pull/4784/commits/f905350f5431351e46841002bdc05b39c9e6cc9c